proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,99 @@
+# **Add `search` Functionality to DotNet**
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add search functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+The `dotnet` CLI is missing a way for customers to search for packages from the command line, as is possible with `nuget.exe search`.
+This brings NuGet experiences in the `dotnet` CLI one step closer towards feature parity.
+
+While [NuGet/Home#6060](https://github.com/NuGet/Home/issues/6060) (the linked issue) also discusses the `nuget.exe list` command, the issue was created before `nuget.exe search` was added.
+The [NuGet Server API Search Resource specification requires all packages to be returned when no `q` parameter is provided](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#request-parameters).
+Therefore, this proposal does not add a `list` equivalent to the `dotnet` CLI, as it is effectively redundant.
+
+## Explanation
+
+### Functional explanation
```

Comment:
Does the search command consider package source mapping settings as well? I'm curious to know how the new search functionality will interact with existing configurations and settings, ensuring that it provides accurate and relevant results based on the user's preferences and restrictions.

Reply:
I am not sure about that. Currently, what I do is go through each of the package sources specified in the config file and do a search API request to each of the sources.

Reply:
Kartheek raises a great question. I believe most CLI search functionality that we have today doesn't honor source mapping. I'm not clear what customers expect to happen. 

Reconciling some of the following thoughts may help reach a decision:
1. It's optional to provide a `--source`. What do customers expect when it's not specified? What if they specify a source and the package isn't found (do we remind them with a warning that source mapping may affect results)?
2. There's no `--ConfigFile` option. If there were, it'd be more support for honoring source mapping.
3. Running a search with no options, do customers expect the context to matter? If my current directory is a solution, should the search's context be the solution by default? Or is search more about user/machine-wide configuration?
4. When the user-level config is applying to the CLI command, ideally there wouldn't be any source mapping in that `nuget.config`.

Reply:
Re: Source mapping 

I'd say for the purposes of this implementation, search is just that, search. I don't think we want source mapping to affect search results, but that's a separate topic that we shouldn't worry about in this context.

Source Mapping is about installing packages. People should be able to discover packages on different feeds in case they want to change mappings or something like that.